.github/workflows/buf-ci.yaml

```
@@ -0,0 +1,25 @@
+name: Buf CI
+on:
+  push:
+  pull_request:
+    types: [opened, synchronize, reopened, labeled, unlabeled]
+  delete:
+permissions:
+  contents: read
+  pull-requests: write
+jobs:
+  buf:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          path: head
+      - uses: actions/checkout@v4
+        with:
+          path: base
+          ref: ${{ github.event.pull_request.base.sha }}
+      - uses: bufbuild/buf-action@v1
+        with:
+          input: head/proto
+          breaking_against: base/proto
```

```suggestion
          input: proto
          breaking_against: 'buf.build/connectrpc/conformance'
```
This would also work, without the path specified in the checkouts above. Reason to prefer using the base branch is for branches on non base branches.

---

It _needs_ to be the BSR for correctness, because these run _after_ commit/push.

So if you only check against the base branch:
1. You inadvertently push an incompatible change (such as a semantic merge conflict: two branches are both compatible with a shared base but not with each other). This check _fails_.
2. You push some other commit later that adds a new field, not yet addressing the compat problem in previous run. Now the check _passes_ because it is only checking compatibility with what's in Git, instead of compatibility with what has actually been published to the BSR. It should fail until a commit is made to _fix_ the compatibility issue identified in step 1.

Since the BSR module is what we recommend people consume and use for code generation, that is the thing where compatibility matters. Previous commit of main (or base branch) isn't really strong enough.

---

Agree, fixed.